pkg/virt-operator/resource/apply/reconcile.go

```diff
@@ -125,15 +125,42 @@ const (
 )
 
 // Merge all Tolerations, Affinity and NodeSelectos from NodePlacement into pod spec
-func InjectPlacementMetadata(componentConfig *v1.ComponentConfig, podSpec *corev1.PodSpec) {
+func InjectPlacementMetadata(componentConfig *v1.ComponentConfig, podSpec *corev1.PodSpec, defaultScheduleToCPNodes bool) {
 	if podSpec == nil {
 		podSpec = &corev1.PodSpec{}
 	}
+
 	if componentConfig == nil || componentConfig.NodePlacement == nil {
-		componentConfig = &v1.ComponentConfig{
-			NodePlacement: &v1.NodePlacement{},
+		if defaultScheduleToCPNodes {
+			componentConfig = &v1.ComponentConfig{
+				NodePlacement: &v1.NodePlacement{
+					Affinity: &corev1.Affinity{
+						NodeAffinity: &corev1.NodeAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+								NodeSelectorTerms: []corev1.NodeSelectorTerm{
+									{
+										MatchExpressions: []corev1.NodeSelectorRequirement{
+											{
+												Key:      "node-role.kubernetes.io/control-plane",
+												Operator: corev1.NodeSelectorOpExists,
+											},
+											{
+												Key:      "node-role.kubernetes.io/worker",
+												Operator: corev1.NodeSelectorOpDoesNotExist,
+											},
```

Comment:
This is wrong, control planes can be workers too, they're still control planes and can still schedule control-plane-related workloads.

Reply:
Right, but then how do we mitigate the security risk? IIUC the whole idea is to not schedule pods into worker nodes since it's dangerous.

What we can do is require scheduling to CP nodes, but prefer not landing on worker nodes.
Any objections?

Reply:
The reason I'm saying that is because there may be nodes on the clusters that are CP only, and some that are CP + worker. I think a sensible default is to prefer not landing on CP+worker nodes.

Reply:
>What we can do is require scheduling to CP nodes, but prefer not landing on worker nodes.
Any objections?

:+1:

Reply:
Preferring non-workers is a good idea.
The most important by far, in my opinion, is to avoid scheduling CP workloads on non-CP nodes.

Reply:
Done :)
PTAL